Sources/PackageCollectionsSigning/Key/Key.swift

```diff
@@ -0,0 +1,63 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2021 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import Foundation
+
+protocol PrivateKey {
+    /// Creates a private key from PEM.
+    ///
+    /// - Parameters:
+    ///   - pem: The key in PEM format, including the `-----BEGIN` and `-----END` lines.
+    init<Data>(pem data: Data) throws where Data: DataProtocol
+}
+
+protocol PublicKey {
+    /// Creates a public key from raw bytes.
+    ///
+    /// Refer to implementation for details on what representation the raw bytes should be.
+    init(data: Data) throws
+
+    /// Creates a public key from PEM.
+    ///
+    /// - Parameters:
+    ///   - pem: The key in PEM format, including the `-----BEGIN` and `-----END` lines.
+    init<Data>(pem data: Data) throws where Data: DataProtocol
+}
+
+enum KeyError: Error {
+    case initializationFailure
+    case invalidPEM
+    case invalidData
+}
+
+enum KeyType {
+    case RSA
```

Comment:
Sent you an email to discuss further. From a security standpoint, we would prefer not supporting RSA for this use case as there doesn't seem to be a backwards compatibility requirement.

Reply:
Per my email reply, we need to support RSA.

Reply:
I added key size check to https://github.com/apple/swift-package-manager/pull/3242 (https://github.com/apple/swift-package-manager/pull/3242/commits/ecaec8a1441c17dd222cce0d7029441dc324a056) since I can't find a good spot to do that in this PR.